lib/instrumentation/modules/aws-sdk/sns.js

```diff
@@ -0,0 +1,102 @@
+const constants = require('../../../constants')
+
+const TYPE = 'messaging'
+const SUBTYPE = 'sns'
+const ACTION = 'PUBLISH'
+const ACCESS_POINT = 'accesspoint'
+const PHONE_NUMBER = '<PHONE_NUMBER>'
+
+function getRegionFromRequest (request) {
+  return request && request.service &&
+        request.service.config && request.service.config.region
+}
+
+function getSpanNameFromRequest (request) {
+  const topicName = getDestinationNameFromRequest(request)
+  return `SNS ${ACTION} ${topicName}`
+}
+
+function getMessageContextFromRequest (request) {
+  return {
+    queue_name: getDestinationNameFromRequest(request)
+  }
+}
+
+function getMessageDestinationContextFromRequest (request) {
+  return {
+    resource: `${SUBTYPE}/${getDestinationNameFromRequest(request)}`,
+    type: TYPE,
+    name: SUBTYPE,
+    cloud: {
+      region: getRegionFromRequest(request)
+    }
+  }
+}
+
+function getDestinationNameFromRequest (request) {
+  const phoneNumber = request && request.params && request.params.PhoneNumber
+  if (phoneNumber) {
+    return PHONE_NUMBER
+  }
+
+  let arn = request && request.params && request.params.TopicArn
+  if (!arn) {
+    arn = request && request.params && request.params.TargetArn
+  }
+
+  if (!arn) {
+    return
+  }
+  const parts = arn.split(':')
+  const topicOrAccessPointName = parts.pop()
+  const maybeAccessPoint = parts.pop()
+  if (ACCESS_POINT === maybeAccessPoint) {
+    return `${ACCESS_POINT}:${topicOrAccessPointName}`
+  }
+  return topicOrAccessPointName
```

Comment:
I have some comments/concerns for the specification here: https://github.com/elastic/apm/pull/421#discussion_r673542576

If we are not in a hurry to merge this, perhaps we should wait for that discussion. We can come back to it as well if that discussion lingers.

Reply:
Sounds good to me -- the other changes have been made and are ready to go.  I'll leave this one open for now and check back in periodically.

Reply:
This change moved to https://github.com/elastic/apm/pull/478

Reply:
https://github.com/elastic/apm/pull/478 is merged now, so this case be complete when you have a chance again.